collect_app/src/main/java/org/odk/collect/android/tasks/DownloadFormListTask.java

```diff
@@ -405,30 +405,28 @@ private boolean isNewerFormVersionAvailable(String md5Hash) {
     }
 
     private boolean areNewerMediaFilesAvailable(String formId, String formVersion, List<MediaFile> newMediaFiles) {
-        String mediaDirPath = new FormsDao().getFormMediaPath(formId, formVersion);
-        return !areMediaFilesIdentical(newMediaFiles, mediaDirPath);
-    }
-
-    private boolean areMediaFilesIdentical(List<MediaFile> newMediaFiles, String mediaDirPath) {
-        boolean result = true;
+        boolean result = false;
 
         if (newMediaFiles != null) {
+            String mediaDirPath = new FormsDao().getFormMediaPath(formId, formVersion);
             List<File> localMediaFiles = FileUtils.getAllFormMediaFiles(mediaDirPath);
             if (localMediaFiles != null) {
-                if (localMediaFiles.size() == newMediaFiles.size()) {
-                    for (MediaFile mediaFile : newMediaFiles) {
-                        File file = getFileByName(localMediaFiles, mediaFile.getFilename());
+                for (MediaFile mediaFile : newMediaFiles) {
+                    File file = getFileByName(localMediaFiles, mediaFile.getFilename());
+                    if (file != null) {
                         String mediaFileHash = mediaFile.getHash();
                         mediaFileHash = mediaFileHash.substring(4, mediaFileHash.length());
                         if (!mediaFileHash.equals(FileUtils.getMd5Hash(file))) {
-                            result = false;
+                            result = true;
+                            break;
```

Comment:
`return true` makes the intent more clear here. Same below -- all `true` cases can return immediately and then the false case can be at the end. Then I don't think you even need the `result` variable.